Integration test fails for http filter example

Command 
```bash
$ bazel test //http-filter-example:http_filter_integration_test 
```

Output
```
INFO: Analysed target //http-filter-example:http_filter_integration_test (0 packages loaded).
INFO: Found 1 test target...
ERROR: /home/june/git/envoy-filter-example/http-filter-example/BUILD:47:1: C++ compilation of rule '//http-filter-example:http_filter_integration_test_lib' failed (Exit 1)
http-filter-example/http_filter_integration_test.cc: In member function 'virtual void Envoy::HttpFilterSampleIntegrationTest_Test1_Test::TestBody()':
http-filter-example/http_filter_integration_test.cc:33:57: error: no matching function for call to 'Envoy::IntegrationCodecClient::makeHeaderOnlyRequest(Envoy::Http::TestHeaderMapImpl&, Envoy::IntegrationStreamDecoder&)'
   codec_client->makeHeaderOnlyRequest(headers, *response);
                                                         ^
In file included from http-filter-example/http_filter_integration_test.cc:1:0:
external/envoy/test/integration/http_integration.h:25:31: note: candidate: Envoy::IntegrationStreamDecoderPtr Envoy::IntegrationCodecClient::makeHeaderOnlyRequest(const Envoy::Http::HeaderMap&)
   IntegrationStreamDecoderPtr makeHeaderOnlyRequest(const Http::HeaderMap& headers);
                               ^
external/envoy/test/integration/http_integration.h:25:31: note:   candidate expects 1 argument, 2 provided
http-filter-example/http_filter_integration_test.cc:34:84: error: no matching function for call to 'Envoy::FakeUpstream::waitForHttpConnection(Envoy::Event::Dispatcher&)'
   fake_upstream_connection = fake_upstreams_[0]->waitForHttpConnection(*dispatcher_);
                                                                                    ^
In file included from external/envoy/test/integration/integration.h:11:0,
                 from external/envoy/test/integration/http_integration.h:10,
                 from http-filter-example/http_filter_integration_test.cc:1:
external/envoy/test/integration/fake_upstream.h:512:3: note: candidate: testing::AssertionResult Envoy::FakeUpstream::waitForHttpConnection(Envoy::Event::Dispatcher&, Envoy::FakeHttpConnectionPtr&, std::chrono::milliseconds)
   waitForHttpConnection(Event::Dispatcher& client_dispatcher, FakeHttpConnectionPtr& connection,
   ^
external/envoy/test/integration/fake_upstream.h:512:3: note:   candidate expects 3 arguments, 1 provided
external/envoy/test/integration/fake_upstream.h:524:3: note: candidate: static testing::AssertionResult Envoy::FakeUpstream::waitForHttpConnection(Envoy::Event::Dispatcher&, std::vector<std::unique_ptr<Envoy::FakeUpstream> >&, Envoy::FakeHttpConnectionPtr&, std::chrono::milliseconds)
   waitForHttpConnection(Event::Dispatcher& client_dispatcher,
   ^
external/envoy/test/integration/fake_upstream.h:524:3: note:   candidate expects 4 arguments, 1 provided
http-filter-example/http_filter_integration_test.cc:35:75: error: no matching function for call to 'Envoy::FakeHttpConnection::waitForNewStream(Envoy::Event::Dispatcher&)'
   request_stream = fake_upstream_connection->waitForNewStream(*dispatcher_);
                                                                           ^
In file included from external/envoy/test/integration/integration.h:11:0,
                 from external/envoy/test/integration/http_integration.h:10,
                 from http-filter-example/http_filter_integration_test.cc:1:
external/envoy/test/integration/fake_upstream.h:401:3: note: candidate: testing::AssertionResult Envoy::FakeHttpConnection::waitForNewStream(Envoy::Event::Dispatcher&, Envoy::FakeStreamPtr&, bool, std::chrono::milliseconds)
   waitForNewStream(Event::Dispatcher& client_dispatcher, FakeStreamPtr& stream,
   ^
external/envoy/test/integration/fake_upstream.h:401:3: note:   candidate expects 4 arguments, 1 provided
Target //http-filter-example:http_filter_integration_test failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 320.534s, Critical Path: 225.40s
INFO: 390 processes: 390 linux-sandbox.
FAILED: Build did NOT complete successfully

FAILED: Build did NOT complete successfully

```

--- a/http-filter-example/http_filter_integration_test.cc
+++ b/http-filter-example/http_filter_integration_test.cc
@@ -1,5 +1,4 @@
 #include "test/integration/http_integration.h"
-#include "test/integration/utility.h"
 
 namespace Envoy {
 class HttpFilterSampleIntegrationTest : public HttpIntegrationTest,
@@ -26,13 +25,12 @@ TEST_P(HttpFilterSampleIntegrationTest, Test1) {
 
   IntegrationCodecClientPtr codec_client;
   FakeHttpConnectionPtr fake_upstream_connection;
-  IntegrationStreamDecoderPtr response(new IntegrationStreamDecoder(*dispatcher_));
   FakeStreamPtr request_stream;
 
   codec_client = makeHttpConnection(lookupPort("http"));
-  codec_client->makeHeaderOnlyRequest(headers, *response);
-  fake_upstream_connection = fake_upstreams_[0]->waitForHttpConnection(*dispatcher_);
-  request_stream = fake_upstream_connection->waitForNewStream(*dispatcher_);
+  auto response = codec_client->makeHeaderOnlyRequest(headers);
+  fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection, std::chrono::milliseconds(5));
+  fake_upstream_connection->waitForNewStream(*dispatcher_, request_stream);
   request_stream->waitForEndStream(*dispatcher_);
   response->waitForEndStream();
 

